data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/prepper/peerforwarder/discovery/PeerListProviderFactory.java

```diff
@@ -36,10 +40,15 @@ public PeerListProvider createProvider(final PluginSetting pluginSetting) {
             case STATIC:
                 final List<String> endpoints = (List<String>) pluginSetting.getAttributeOrDefault(PeerForwarderConfig.STATIC_ENDPOINTS, null);
                 Objects.requireNonNull(endpoints, String.format("Missing '%s' configuration value", PeerForwarderConfig.STATIC_ENDPOINTS));
+                Preconditions.checkArgument(endpoints.stream().allMatch(this::validateEndpoint), "Including invalid endpoints: %s", endpoints);
```

Comment:
This won't indicate which endpoints are actually invalid. 

Might be better to stream-create a list of invalid endpoints, assert the list is empty, and if not then display the list contents to the user.